float64 k_gain
float64 d_gain
